Build error: Module Not Found: Can't Resolve '../theme/icons/search.svg' in AI Agent Plugin

#### Summary

Module not found error during the build process for the CKEditor AI Agent plugin. The error indicates that the `../theme/icons/search.svg` file is missing or cannot be resolved.

---

#### Steps to reproduce

1. Clone the `ckeditor_ai_agent` project repository.
2. Run `npm install` to install dependencies.
3. Execute `npm run build`.

---

#### What is the current _bug_ behavior?

The build process fails with the following error:  
`Module not found: Error: Can't resolve '../theme/icons/search.svg'`.

---

#### What is the expected _correct_ behavior?

The build process should complete successfully, including all required files, without any missing asset errors.

---

#### Relevant logs and/or screenshots

```plaintext
ERROR in ./js/ckeditor5_plugins/aiagent/src/aiagentui.js 4:0-51
Module not found: Error: Can't resolve '../theme/icons/search.svg' in '/Users/jur/www/git-projects/dxpr/8/ckeditor_ai_agent/js/ckeditor5_plugins/aiagent/src'
resolve '../theme/icons/search.svg' in '/Users/jur/www/git-projects/dxpr/8/ckeditor_ai_agent/js/ckeditor5_plugins/aiagent/src'
...
webpack 5.96.1 compiled with 1 error in 479 ms
```

---

--- a/build.js
+++ b/build.js
@@ -11,9 +11,45 @@ if (!fs.existsSync(destinationDir)) {
 }
 
 try {
+  // First, remove only the ai-agent.js file from build directory
+  const buildDir = path.join(__dirname, 'js/build');
+  const buildFile = path.join(buildDir, 'ai-agent.js');
+  
+  if (fs.existsSync(buildFile)) {
+    console.log('Removing existing build file...');
+    fs.unlinkSync(buildFile);
+  }
+
+  // Ensure build directory exists
+  if (!fs.existsSync(buildDir)) {
+    fs.mkdirSync(buildDir, { recursive: true });
+  }
+  console.log('Build directory ready.');
+
+  // Copy translations from package build directory
+  const sourceTranslationsDir = path.join(
+    __dirname,
+    'node_modules/@dxpr/ckeditor5-ai-agent/build/translations'
+  );
+  const destTranslationsDir = path.join(buildDir, 'translations');
+
+  if (fs.existsSync(sourceTranslationsDir)) {
+    if (!fs.existsSync(destTranslationsDir)) {
+      fs.mkdirSync(destTranslationsDir, { recursive: true });
+    }
+    copyRecursively(sourceTranslationsDir, destTranslationsDir);
+    console.log('Translations successfully copied.');
+  }
+
+  // Remove existing destination directory
+  if (fs.existsSync(destinationDir)) {
+    console.log('Removing existing destination directory...');
+    fs.rmSync(destinationDir, { recursive: true, force: true });
+  }
+  fs.mkdirSync(destinationDir, { recursive: true });
+  
   console.log('Installing @dxpr/ckeditor5-ai-agent package...');
-  // Install the package locally in a temporary location
-  execSync('npm install @dxpr/ckeditor5-ai-agent --no-save', { stdio: 'inherit' });
+  execSync('npm install @dxpr/ckeditor5-ai-agent', { stdio: 'inherit' });
 
   console.log('Copying files from @dxpr/ckeditor5-ai-agent/src to destination...');
   // Source directory inside the package
@@ -29,16 +65,34 @@ try {
   }
 
   // Copy files recursively from source to destination
-  fs.readdirSync(sourceDir).forEach((file) => {
-    const sourceFile = path.join(sourceDir, file);
-    const destFile = path.join(destinationDir, file);
-
-    // Check if the item is a file and has .js or .ts extension
-    if (fs.lstatSync(sourceFile).isFile() && /\.(js|ts)$/.test(file)) {
-      fs.copyFileSync(sourceFile, destFile);
-      console.log(`Copied: ${file}`);
+  function copyRecursively(source, destination) {
+    // Create destination directory if it doesn't exist
+    if (!fs.existsSync(destination)) {
+      fs.mkdirSync(destination, { recursive: true });
     }
-  });
+
+    // Read source directory
+    const files = fs.readdirSync(source);
+
+    files.forEach(file => {
+      const sourcePath = path.join(source, file);
+      const destPath = path.join(destination, file);
+
+      const stats = fs.lstatSync(sourcePath);
+      
+      if (stats.isDirectory()) {
+        // Recursively copy subdirectories
+        console.log(`Creating directory: ${destPath}`);
+        copyRecursively(sourcePath, destPath);
+      } else {
+        // Copy all files
+        fs.copyFileSync(sourcePath, destPath);
+        console.log(`Copied file: ${sourcePath} -> ${destPath}`);
+      }
+    });
+  }
+
+  copyRecursively(sourceDir, destinationDir);
 
   console.log('Files successfully copied to destination directory.');
 
@@ -79,37 +133,20 @@ try {
     console.warn('index.js not found, skipping modification.');
   }
 
-  // Copy theme directory
+  // Update theme directory copying to handle icons subdirectory
   const sourceThemeDir = path.join(
     __dirname,
     'node_modules/@dxpr/ckeditor5-ai-agent/theme'
   );
   const destThemeDir = path.join(__dirname, 'js/ckeditor5_plugins/aiagent/theme');
 
-  if (!fs.existsSync(destThemeDir)) {
-    fs.mkdirSync(destThemeDir, { recursive: true });
-  }
-
-  // Copy theme files recursively
-  function copyDir(src, dest) {
-    fs.readdirSync(src).forEach((item) => {
-      const srcPath = path.join(src, item);
-      const destPath = path.join(dest, item);
-
-      if (fs.lstatSync(srcPath).isDirectory()) {
-        if (!fs.existsSync(destPath)) {
-          fs.mkdirSync(destPath, { recursive: true });
-        }
-        copyDir(srcPath, destPath);
-      } else {
-        fs.copyFileSync(srcPath, destPath);
-        console.log(`Copied: ${item}`);
-      }
-    });
-  }
-
   if (fs.existsSync(sourceThemeDir)) {
-    copyDir(sourceThemeDir, destThemeDir);
+    if (!fs.existsSync(destThemeDir)) {
+      fs.mkdirSync(destThemeDir, { recursive: true });
+    }
+    
+    // Copy theme files and maintain directory structure
+    copyRecursively(sourceThemeDir, destThemeDir);
     console.log('Theme directory successfully copied.');
   } else {
     console.warn('Theme directory not found in source package.');

